File: src/main/webapp/app/items/CodeType.js
(function() {
   "use strict";

   angular.module("App")
      .constant("CodeType", createEnum());

   function createEnum() {
      return {
         GTIN: new CodeType("GTIN"),
         UPC: new CodeType("UPC"),
         NACS: new CodeType("NACS"),
         MANUAL: new CodeType("Manual"),
         RANDOM: new CodeType("Random")
      };
   }

   function CodeType(displayString) {
      this.displayString = displayString;
   }
})();

File: src/main/java/org/codegas/itemmanagement/domain/value/CodeType.java
package org.codegas.itemmanagement.domain.value;

public enum CodeType {
    GTIN("%013d"),
    UPC("%012d"),
    PLU("%05d"),
    NACS("%02d%02d%02d"),
    MANUAL("%s"),
    RANDOM("%s");

    private final String valueFormat;

    CodeType(String valueFormat) {
        this.valueFormat = valueFormat;
    }

    public String getValueFormat() {
        return valueFormat;
    }
}
